subworkflows/local/circrna_discovery.nf

```diff
@@ -0,0 +1,227 @@
+
+    mate2 = reads.map{ meta, reads -> return [ meta, reads[1] ] }
+    DCC_MATE2_1ST_PASS( mate2, star_index, gtf, true, '', '' )
+    DCC_MATE2_SJDB( DCC_MATE2_1ST_PASS.out.tab.map{ meta, tab -> return [ tab ] }.collect(), bsj_reads )
+    DCC_MATE2_2ND_PASS( mate2, star_index, DCC_MATE2_SJDB.out.sjtab, true, '', '' )
+
+    dcc_stage = DCC_2ND_PASS.out.junction.join( DCC_MATE1_2ND_PASS.out.junction, remainder: true ).join( DCC_MATE2_2ND_PASS.out.junction, remainder: true )
+    dcc = dcc_stage.map{ it -> def meta = it[0]; if( meta.single_end ){ return [ it[0], it[1], [], [] ] } else { return it } }.view()
+    DCC( dcc, fasta, gtf )
+    DCC_FILTER( DCC.out.txt.map{ meta, txt -> meta.tool = "dcc"; return [ meta, txt ] }, bsj_reads )
+
+    ch_versions = ch_versions.mix(DCC_MATE1_1ST_PASS.out.versions)
+    ch_versions = ch_versions.mix(DCC.out.versions)
+
+    //
+    // MAPSPLICE WORKFLOW:
+    //
+
+    MAPSPLICE_REFERENCE( gtf )
+    MAPSPLICE_ALIGN( reads, bowtie_index.collect(), chromosomes, gtf )
+    MAPSPLICE_PARSE( MAPSPLICE_ALIGN.out.raw_fusions )
+    MAPSPLICE_ANNOTATE( MAPSPLICE_PARSE.out.junction, fasta, MAPSPLICE_REFERENCE.out.txt )
+    mapsplice_filter = MAPSPLICE_ANNOTATE.out.txt.map{ meta, txt -> meta.tool = "mapsplice"; return [ meta, txt ] }
+    MAPSPLICE_FILTER( mapsplice_filter, bsj_reads )
+
+    ch_versions = ch_versions.mix(MAPSPLICE_REFERENCE.out.versions)
+    ch_versions = ch_versions.mix(MAPSPLICE_ALIGN.out.versions)
+    ch_versions = ch_versions.mix(MAPSPLICE_PARSE.out.versions)
+    ch_versions = ch_versions.mix(MAPSPLICE_ANNOTATE.out.versions)
+
+    //
+    // ANNOTATION WORKFLOW:
+    //
+
+    circrna_filtered = CIRCEXPLORER2_FILTER.out.results.mix(SEGEMEHL_FILTER.out.results, CIRCRNA_FINDER_FILTER.out.results, FIND_CIRC_FILTER.out.results, CIRIQUANT_FILTER.out.results, DCC_FILTER.out.results, MAPSPLICE_FILTER.out.results )
+    ANNOTATION( circrna_filtered, gtf, exon_boundary )
+
+    ch_versions = ch_versions.mix(ANNOTATION.out.versions)
+
+    //
+    // FASTA WORKFLOW:
+    //
+
+    FASTA( ANNOTATION.out.bed, fasta )
+
+    ch_versions = ch_versions.mix(FASTA.out.versions)
+
+    //
+    // COUNT MATRIX WORKFLOW:
+    //
+
+    ch_matrix = CIRCEXPLORER2_FILTER.out.matrix.mix(SEGEMEHL_FILTER.out.matrix, CIRCRNA_FINDER_FILTER.out.matrix, FIND_CIRC_FILTER.out.matrix, CIRIQUANT_FILTER.out.matrix, DCC_FILTER.out.matrix, MAPSPLICE_FILTER.out.matrix )
+    tools_selected = params.tool.split(',').collect{it.trim().toLowerCase()}
+
+    if( tools_selected.size() > 1){
+
+        MERGE_TOOLS( ch_matrix.map{ meta, bed -> var = [:]; var.id = meta.id; return [ var, bed ] }.groupTuple(), tool_filter, duplicates_fun )
+
+        COUNTS_COMBINED( MERGE_TOOLS.out.merged.map{ meta, bed -> return [ bed ] }.collect() )
+
+        dea_matrix = COUNTS_COMBINED.out.dea_matrix
+        clr_matrix = COUNTS_COMBINED.out.clr_matrix
+
+    }else{
+
+        // TODO: concerned that this does not wait for all files?
```

Comment:
Has this been checked? Never used it myself, but maybe [this operator](https://www.nextflow.io/docs/latest/operator.html#buffer)can help some how?

But I believe `groupTuple` will wait until everything is in.

Reply:
Remnant from dev tests, works now